pkg/hostname/hostname.go

```diff
@@ -10,32 +10,62 @@ import (
 	"github.com/1898andCo/HAOS/pkg/config"
 )
 
-func SetHostname(c *config.CloudConfig) error {
+type Abstract interface {
+	Sethostname([]byte) error
+	Hostname() (string, error)
+	Open(string) (*os.File, error)
+	Close(*os.File) error
```

Comment:
may be able to avoid this method in the interface - it can just run on the *os.File return from Open()